src/tools/compiletest/src/runtest.rs

```diff
@@ -2467,6 +2467,8 @@ impl<'test> TestCx<'test> {
                     "-Zvalidate-mir",
                     "-Zlint-mir",
                     "-Zdump-mir-exclude-pass-number",
+                    "--crate-type=rlib",
+                    "-Clink-dead-code",
```

Comment:
Nit: Can we comment why we're passing `-Clink-dead-code`? I'm sure eventually someone will want to test that we only generate MIR for live code or something and be confused why we've set this option.

Reply:
Only one test _needs_ `-Clink-dead-code`, and I've added a comment to it. I'm planning on doing a follow-up PR to clean up the mir-opt tests as a result of changing the default crate-type. A handful rely on `fn main` being reachable, I've patched those up because `-Clink-dead-code` was papering over that. But there are a handful that set the crate-type to lib/rlib... because that's just a better default for writing these tests.

Reply:
I like that better, thanks!